tests/test_datasets/test_dataset_functions.py

```diff
@@ -1341,47 +1341,34 @@ def test_get_dataset_cache_format_feather(self):
         self.assertEqual(len(attribute_names), X.shape[1])
 
     def test_data_edit(self):
-
-        # admin key for test server (only admins or owners can edit datasets).
-        # all users can edit their own datasets)
-        openml.config.apikey = "d488d8afd93b32331cf6ea9d7003d4c3"
-
-        # case 1, editing description, creator, contributor, collection_date, original_data_url,
-        # paper_url, citation, language edits existing dataset.
+        # Case 1
+        # All users can edit non-critical fields of datasets
+        desc = "xor dataset representing XOR operation"
         did = 564
         result = edit_dataset(
             did,
-            description="xor dataset represents XOR operation",
-            contributor="",
+            description=desc,
+            contributor="xxx",
             collection_date="2019-10-29 17:06:18",
             original_data_url="https://www.kaggle.com/ancientaxe/and-or-xor",
             paper_url="",
             citation="kaggle",
             language="English",
         )
-        self.assertEqual(result, did)
-
-        # case 2, editing data, attributes, default_target_attribute, row_id_attribute,
-        # ignore_attribute generates a new dataset
+        self.assertEqual(did, result)
+        edited_dataset = openml.datasets.get_dataset(did)
+        self.assertEqual(edited_dataset.description, desc)
 
-        column_names = [
-            ("input1", "REAL"),
-            ("input2", "REAL"),
-            ("y", "REAL"),
-        ]
+        # Case 2
+        # only admins or owners can edit all critical fields of datasets
+        # admin key for test server
+        openml.config.apikey = "d488d8afd93b32331cf6ea9d7003d4c3"
```

Comment:
I had somehow overlooked this previously. Is it possible to generate a new admin API token? I think we should be able to configure the api key as an environment variable through our travis/appveyor accounts so that we don't need to expose this high access api key.

Reply:
> Do we want to allow everyone to edit the data? Is this temporary? I can't imagine this is healthy long term, someone will come along and be destructive.
> 
> Of course we do still need a way for other users to at least _suggest_ changes to be approved by the original uploader or someone from openml (hopefully mostly the former). Are there plans for this?

Yes this is temporary and we have plans to modify this in future such that the owner has some control. Owners can choose to allow or not allow community edits for non-critical features in future. For reference, here is the whole dicussion  openml/OpenML#1058.

Reply:
> I had somehow overlooked this previously. Is it possible to generate a new admin API token? I think we should be able to configure the api key as an environment variable through our travis/appveyor accounts so that we don't need to expose this high access api key.

Instead I could change this test such that this admin API key is not required. The test is for whether an  owner(or admin) can edit a critical feature, so it can be changed to edit a dataset owned by CI.

Reply:
> The goal of the test is primarily to test whether an owner(or ofcourse admin) can edit a critical feature, so it can be changed to edit a dataset owned by CI (eg: 315) or Test Test (eg: 564)

I would prefer that (and reset the admin API key after - it should be considered public right now).

Reply:
> I think we should be able to configure the api key as an environment variable through our travis/appveyor accounts so that we don't need to expose this high access api key.

That does not work for PRs from outside the OpenML organization. Such secrets will be visible to the PR, and if someone does a PR, that person has access to the secret. It would make it less obvious though that we use such a secret. I'm also not sure what to do with local tests then.